ql/src/java/org/apache/hadoop/hive/ql/cache/results/QueryResultsCache.java

```diff
@@ -631,13 +632,13 @@ public long getSize() {
     }
   }
 
-  public void notifyTableChanged(String dbName, String tableName, long updateTime) {
-    LOG.debug("Table changed: {}.{}, at {}", dbName, tableName, updateTime);
+  public void notifyTableChanged(TableName tableName, long updateTime) {
+    LOG.debug("Table changed: {}, at {}", tableName.getNotEmptyDbTable(), updateTime);
     // Invalidate all cache entries using this table.
     List<CacheEntry> entriesToInvalidate = null;
     rwLock.writeLock().lock();
     try {
-      String key = (dbName.toLowerCase() + "." + tableName.toLowerCase());
+      String key = (tableName.getNotEmptyDbTable().toLowerCase());
```

Comment:
we might want to consider to remove all these "toLowerCase"  calls;and instead make it a contract for tablenames; so it's enforced at the time the tablename is created